typings/context.d.ts

```diff
@@ -0,0 +1,480 @@
+import * as tt from './telegram-types'
+
+import { AdminPerms, Telegram } from './telegram'
+
+export declare class ContextMessageUpdate {
```

Comment:
I still think this should be renamed to Context internally as the JavaScript Code calls it Context.

The name `ContextMessageUpdate` should only be in `export {Context as ContextMessageUpdate}` and thats only for legacy reasons.
As you pointed out correctly the name does not exist in the JavaScript Code so it shouldnt exist in TypeScript either.

Reply:
I agree. However I'm worried about making this PR unreviewable (and thus unmergable) by changing too much.

Reply:
`SceneContextMessageUpdate` would have to be renamed as well, probably to `SceneContext`.

Reply:
I think reviewing coming from JavaScript is easier when looking into each file at a time. Comparing `context.js` with `context.d.ts` and so on. Having the same names only makes it easier to do so.

As this is splitting one file onto many it is already hard to review just by looking into the changes.

Looking into specific commits doing only small things also helps a lot.

Renaming Types does not change much from the functionality side of view so I think it should be done now in a bunch. Then only this PR is breaking and requiring manual work to do after updating to it.

Reply:
I just realized `Context` or `ContextMessageUpdate` is named `TelegrafContext` in the Sourcecode
https://github.com/telegraf/telegraf/blob/646476197d2c6b7e02bea0ba6a9f03da67ac2809/context.js#L53